api/api/controllers/elasticsearch/helpers.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import functools
+import logging as log
+import pprint
+import time
+
+from django.conf import settings
+
+from elasticsearch import BadRequestError, NotFoundError
+
+
+def log_timing_info(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        start_time = time.time()
+
+        # Call the original function
+        result = func(*args, **kwargs)
+
+        response_time_in_ms = int((time.time() - start_time) * 1000)
+        if hasattr(result, "took"):
+            es_time_in_ms = result.took
+        else:
+            es_time_in_ms = result["took"]
+        log.info(
+            {
+                "response_time": response_time_in_ms,
+                "es_time": es_time_in_ms,
+                "search_query": kwargs.get("search_query"),
+            }
+        )
+
+        return result
+
+    return wrapper
+
+
+@log_timing_info
+def get_es_response(s, search_query=None):
+    if settings.VERBOSE_ES_RESPONSE:
+        log.info(pprint.pprint(s.to_dict()))
+
+    try:
+        search_response = s.execute()
+
+        if settings.VERBOSE_ES_RESPONSE:
+            log.info(pprint.pprint(search_response.to_dict()))
+    except (BadRequestError, NotFoundError) as e:
+        raise ValueError(e)
+
+    return search_response
```

Comment:
Just a nit, but wrapping all of this technically also times how long it takes to write the logs. Not a big deal, that time is probably negligible, but worth mentioning in case somehow it ends up mattering down the line.